app/src/main/java/com/sun_asterisk/moviedb_50/screen/listmovie/ListMovieFragment.kt

```diff
@@ -19,27 +41,145 @@ class ListMovieFragment : Fragment(), ListMovieContract.View {
                 MovieLocalDataSource
             )
         presenter = ListMoviePresenter(movieRepository)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_list_movie, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initToolBar()
         initPresenter()
+        initAdapter()
     }
 
     override fun onGetMoviesSuccess(movies: List<Movie>) {
-        TODO("Not yet implemented")
+        if (adapter.itemCount == 0) {
+            adapter.updateData(movies)
+        } else {
+            adapter.insertData(movies)
+        }
+    }
+
+    override fun onGetMovieResultPage(movieResultPage: MovieResultPage) {
+        totalPage = movieResultPage.movieTotalPage
+        view?.run {
+            val result =
+                getString(R.string.results_found) +
+                        movieResultPage.movieTotalResult
+            resultTextView.text = result
+
+        }
     }
 
     override fun onError(exception: Exception?) {
-        TODO("Not yet implemented")
+        exception?.let {
+            Toast.makeText(activity, it.message.toString(), Toast.LENGTH_LONG).show()
+        }
     }
 
     override fun onLoading(isLoad: Boolean) {
-        TODO("Not yet implemented")
+        view?.run {
+            if (!isLoad) {
+                listMovieProgressBarLayout.visibility = View.VISIBLE
+            } else {
+                listMovieProgressBarLayout.visibility = View.GONE
+                loadingProgressBar.visibility = View.GONE
+            }
+        }
     }
 
     private fun initPresenter() {
-        presenter?.setView(this)
+        presenter.setView(this)
         arguments?.run {
             val type = getString(Constant.BASE_TYPE)
             val value = getString(Constant.BASE_VALUE)
-            if (type != null && value != null) presenter?.getMovies(type, value)
+            if (type != null && value != null) presenter.getMovies(
+                type,
+                value,
+                Constant.BASE_PAGE_DEFAULT
+            )
+        }
+    }
+
+    private fun initAdapter() {
+        view?.run {
+            listMovieRecyclerView.setHasFixedSize(true)
+            listMovieRecyclerView.adapter = adapter.apply {
+                onItemClick = { item, _ ->
+                    addFragment(item)
+                }
+            }
+            listMovieRecyclerView.addOnScrollListener(object : OnScrollListener() {
+                override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+                    super.onScrollStateChanged(recyclerView, newState)
+                    if (newState == AbsListView.OnScrollListener.SCROLL_STATE_TOUCH_SCROLL) {
+                        isScrolling = true
+                    }
```

Comment:
viết thế này có đúng logic không ? 
```suggestion
                    isScrolling = 
                        newState == AbsListView.OnScrollListener.SCROLL_STATE_TOUCH_SCROLL
```